lib/qless/middleware/retry_exceptions.rb

```diff
@@ -48,19 +48,18 @@ def on_retry_callback
         @on_retry_callback ||= DEFAULT_ON_RETRY_CALLBACK
       end
 
-      def exponential(base, options = {})
+      # If `factor` is omitted it is set to `delay_seconds` to reproduce legacy
+      # behavior.
+      def exponential(delay_seconds, options={})
+        factor = options.fetch(:factor, delay_seconds)
         fuzz_factor = options.fetch(:fuzz_factor, 0)
 
-        lambda do |num, _error|
-          unfuzzed = base**num
-
-          fuzz = 0
-          unless fuzz_factor.zero?
-            max_fuzz = unfuzzed * fuzz_factor
-            fuzz = rand(max_fuzz) * [1, -1].sample
-          end
-
-          unfuzzed + fuzz
+        lambda do |retry_no, error|
+          unfuzzed = delay_seconds * factor**(retry_no - 1)
+          return unfuzzed if fuzz_factor.zero?
+          r = 2 * rand  - 1
+          # r is uniformly distributed in range [-1, 1]
```

Comment:
Thanks, this is better.  I also just remembered that `rand` can take a range argument, so this could be `r = rand(-1.0..1.0)`, which IMO is more intention revealing.

``` irb
irb(main):001:0> rand(-1.0..1.0)
=> -0.2940561611637065
irb(main):002:0> rand(-1.0..1.0)
=> -0.1088545949516877
irb(main):003:0> rand(-1.0..1.0)
=> 0.32749055093974433
irb(main):004:0> rand(-1.0..1.0)
=> 0.053600710077926994
irb(main):005:0> rand(-1.0..1.0)
=> 0.43536664510968537
```

`